compose/compiler/compiler-hosted/src/main/java/androidx/compose/compiler/plugins/kotlin/lower/decoys/CreateDecoysTransformer.kt

```diff
@@ -178,14 +347,21 @@ class CreateDecoysTransformer(
     @OptIn(ObsoleteDescriptorBasedAPI::class)
     private fun IrFunction.copyWithName(
         newName: Name,
-        factory: (IrFunctionBuilder.() -> Unit) -> IrFunction = context.irFactory::buildFun
+        factory: (IrFunctionBuilder.() -> Unit) -> IrFunction = context.irFactory::buildFun,
+        externalStub: Boolean = false
     ): IrFunction {
         val original = this
         val newFunction = factory {
             updateFrom(original)
             name = newName
             returnType = original.returnType
-            isPrimary = (original as? IrConstructor)?.isPrimary ?: false
+            // In konan backend for K/N, only a primary constructor can has
+            // IrInstanceInitializerCall inside, others MUST have IrDelegatingConstructorCall.
+            // So if we building a decoy implementation constructor, we should check and transfer
+            // "isPrimary" flag if need.
+            isPrimary =
```

Comment:
I try to understand this change:
previous version: `isPrimary = (original as? IrConstructor)?.isPrimary ?: false`
new version `if (original is IrConstructor && context.platform.isNative()) original.isPrimary else false`

If I understand correctly, the previous version did the same but for all targets (k/js, k/n), right? And the new version will set `true` only for k/n. So can we keep the old version for setting `isPrimary` here?